Procedural Dungeon Generator

```diff
@@ -38,10 +52,45 @@ not so close as to be unpassable.
 Add Details
 ==============================
       
-        
+        Now the map needs smaller details to help flesh it out. This is when you add in all
+of the objects that will be scattered about the map. Place tables and chairs in rooms that 
+require them. Create specific items that fit the area they're in, such as dangerous chemicals 
+in a science lab or bunks in a military camp. Computer or electrical panels are also very 
+important. They allow the players to interface with the map in a unique way, and in some 
+cases can be pivotal to the story you are trying to play. When building a map, it's usually 
+a good idea to keep it fairly realistic, but the most important thing to remember is that the 
+players must have fun and that the game be interesting. 
+
+        It's important to write down small details of a map before playing it. It doesn't need 
+to be too in-depth, but it needs to be enough that the GM can improv something interesting, 
+depending on what the players do. 
+
+    For Example: 
+    *  A computer terminal may have access to certain areas of a building, but be locked out of 
+others. It would also be a good idea to think of how the system is secured, as players 
+are able to infiltrate the network.
+    *  A heavily reinforced door might be accessible if the players find the correct wiring in 
+the walls. In general, it's a good idea to have multiple ways for the players to solve a 
+problem. If the there is only one route through your game, it might start to get boring.
+    *  A scrap of paper tacked onto a message board might have the keycode to access a safe 
+where the players might be able to find some loot.
+    *  If you create something that doesn't have rules written for it, feel free to write your 
+own for the encounter. If the enemies in your game have been creating a light-activated, face-
+melting, explosive poison gas, then it's your job to determine exactly how it works.
     
 ==============================
 Add NPCs and Hostiles
 ==============================
     
-        
+        Lastly, your dungeon should have non-playable characters in it. For the sake of interesting 
+story, it's usually a good idea for your characters to have some kind of backstory, whether it's 
+as simple as the organization they work for, or as complicated as the entire detailed story of their 
+life. Combat personnel should be given weapons and armor based off of their levels, and depending on
+the character, you can also give them other skills and feats to go along with their class. Exactly what
+kind of gear they are given is very much up to you and how you want to run your dungeon, but as a 
+rule of thumb, it's a good idea to make combat fast-paced, tactical, and enjoyable for everybody. 
+
+
```

Comment:
... Where do you talk about loot? You hint at puzzles a couple of times, but don't really explicitly call them out. https://www.youtube.com/watch?v=XWgc20zbRXk&list=PLDPL7seU9mMcBZ8yonxZvyEcFHg8dy1wl